src/raftstore/coprocessor/split_observer.rs

```diff
@@ -79,19 +79,24 @@ impl SplitObserver {
                     Ok(key) => {
                         if last_valid_key.as_ref().map_or(false, |k| *k >= key) {
                             warn!(
-                                "[region {}] key {} is not larger than previous {} at {}, skip.",
-                                region_id,
-                                escape(&key),
-                                escape(last_valid_key.as_ref().unwrap()),
-                                k
+                                "key is not larger than previous, skip.";
+                                "region_id" => region_id,
+                                "key" => hex::encode_upper(key),
+                                "previous" => hex::encode_upper(last_valid_key.as_ref().unwrap()),
```

Comment:
Please refer to this PR https://github.com/tikv/tikv/pull/4087.

Reply:
the `Display` trait is implemented for `Key`, but here is `Vec<u8>`. Should I replace `Vec<u8>` with `Key`?

Reply:
No you should use facilities in log_wrapper.